mlrun/runtimes/sparkjob/abstract.py

```diff
@@ -451,16 +462,46 @@ def _run(self, runobj: RunObject, execution: MLClientCtx):
             update_in(job, "spec.mainApplicationFile", self.spec.command)
 
         verify_list_and_update_in(job, "spec.arguments", self.spec.args or [], str)
-        self._submit_job(job, meta.namespace)
+        self._submit_job(job, meta.namespace, code)
 
         return None
 
     def _enrich_job(self, job):
         raise NotImplementedError()
 
-    def _submit_job(self, job, namespace=None):
+    def _submit_job(self, job, namespace=None, code=None):
         k8s = self._get_k8s()
         namespace = k8s.resolve_namespace(namespace)
+        if code:
+            k8s_config_map_name = f"{self.metadata.name}-{uuid.uuid4().hex[:8]}"
+            k8s_config_map = client.V1ConfigMap()
+            k8s_config_map.metadata = client.V1ObjectMeta(
+                name=k8s_config_map_name,
+                namespace=namespace,
+                labels=get_in(job, "metadata.labels"),
+            )
+            k8s_config_map.data = {self.code_script: code}
+            config_map = k8s.v1api.create_namespaced_config_map(
+                namespace, k8s_config_map
+            )
+            config_map_name = config_map.metadata.name
+            from kubernetes import client as k8s_client
+
+            vol_src = client.V1ConfigMapVolumeSource(name=config_map_name)
+            volume_name = "script"
+            vol = client.V1Volume(name=volume_name, config_map=vol_src)
+            vol_mount = k8s_client.V1VolumeMount(
+                mount_path=self.code_path, name=volume_name
+            )
+            update_in(job, "spec.volumes", [vol], append=True)
+            update_in(job, "spec.driver.volumeMounts", [vol_mount], append=True)
+            update_in(job, "spec.executor.volumeMounts", [vol_mount], append=True)
```

Comment:
You're passing objects here, but it should dicts no ? are you sure that's working ?